starter-project-backend/src/tests/__tests__/UserController.test.ts

```diff
@@ -0,0 +1,128 @@
+import User, { UserI } from "../../models/User";
+let chai = require("chai");
+let chaiHttp = require("chai-http");
+let server = require("../../server");
+chai.use(chaiHttp);
+let mongoose = require("mongoose");
+import * as dbHandler from "../setupdb";
+
+describe("User API", () => {
+  let user: UserI;
+  let user2: UserI;
+
+  beforeAll(async () => {
+    await dbHandler.connect();
+  });
+
+  beforeEach(async () => {
+    user = new User({
+      _id: mongoose.Types.ObjectId(),
+      username: `${Date.now().toString()} ${Math.random()}`,
+      password: "$2a$10$efmxm5o1v.inI.eStGGxgO1zHk.L6UoA9LEyYrRPhWkmTQPX8.NKO",
+      firstName: "Abenezer",
+      lastName: "Belay",
+    });
+    await user.save();
+
+    user2 = new User({
+      _id: mongoose.Types.ObjectId(),
+      username: `${Date.now().toString()} ${Math.random()}`,
+      password: "$2a$10$efmxm5o1v.inI.eStGGxgO1zHk.L6UoA9LEyYrRPhWkmTQPX8.NKO",
+      firstName: "Kebede",
+      lastName: "Abebe",
+    });
+
+    await user2.save();
+  });
+
+  afterEach(async () => {
+    await User.findByIdAndDelete(user._id);
+    await User.findByIdAndDelete(user2._id);
+  });
+
```

Comment:
better declare beforeEach, afterEach, before all blocks outside of a test suit because they will be called once in the whole test being running